### PR TITLE
Adding slack webhook task.

### PR DESCRIPTION
**Pull request by chromalley**

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -37,9 +37,12 @@ task {
     certificate-password = ${?APNS_CERTIFICATE_PWD}
     voip-certificate-path = ${?APNS_VOIP_CERTIFICATE_PATH}
     voip-certificate-password = ${?APNS_VOIP_CERTIFICATE_PWD}
+    ua-key = ""
```
Comment (chromalley) [inline src/main/resources/reference.conf]: Prevents an error if configs are not present, but push task is not in use.

File: src/main/scala/co/adhoclabs/task/Config.scala [hunk 2]
```diff
@@ -30,28 +31,16 @@ private[task] trait Config {
   val pushConfig = taskConfig.getConfig("push")
 
   val (apnsCertificatePath, apnsCertificatePwd) = if (consumers.contains(TaskType.inactivePushToken)) {
-    val path = pushConfig.getString("certificate-path")
-    val pass = pushConfig.getString("certificate-password")
-    if (path.isEmpty) {
```
Comment (yeghishe) [inline src/main/scala/co/adhoclabs/task/Config.scala]: What's the advantage of having it "" as default and throwing exception runtime vs not having a default and erring out (I'm assuming with missing key exception)?

File: src/main/scala/co/adhoclabs/task/Config.scala [hunk 3]
```diff
@@ -63,14 +52,28 @@ private[task] trait Config {
   val gcmBaseUrl = pushConfig.getString("gcm-base-url")
   val gcmRequestTimeout = pushConfig.getDuration("request-timeout", MILLISECONDS)
 
+  private def getKeyOrThrow(key: String) = {
+    val config = pushConfig.getString(key)
+    if (!config.isEmpty) key else throw new RuntimeException(s"$key is missing")
```
Comment (yeghishe) [inline src/main/scala/co/adhoclabs/task/Config.scala]: `if (config.isEmpty) else throw new RuntimeException(s"$key is missing") else key`

File: src/main/scala/co/adhoclabs/task/Config.scala [hunk 4]
```diff
@@ -63,14 +52,28 @@ private[task] trait Config {
...
+
   private def taskTypesFromKey(key: String): List[TaskType] =
     taskConfig.getStringList(key).asScala.toList.map(TaskType.fromName(_).getOrElse(throw configError))
 
-  val workerActorPropMap = Map(
-    TaskType.segment -> SegmentActor.props(SegmentClient),
-    TaskType.push -> PushActor.props(UAClient, GcmClient),
-    TaskType.voipPush -> VoipPushActor.props()
-  )
+  val workerActorPropMap = consumers.filterNot(_ == TaskType.demo).map { t ⇒
```
Comment (yeghishe) [inline src/main/scala/co/adhoclabs/task/Config.scala]: Other than filterNot how is this different from normal map? Also why are we filtering out demo?

**Pull request closed without merging**